### DISCUSSION TITLE
Spin `mungectl` out into its own repository

### DISCUSSION THREAD

**Discussion by NucciTheBoss** [canonical/hpc-team#1] [General]
## Motivation

Currently, the Slurm snap contains an embedded utility named `mungectl`: https://github.com/charmed-hpc/slurm-snap/tree/main/src/mungectl

We created `mungectl` with the idea of simplifying munge operations like managing keys, and be able to use it a variety of situations such as within snaps and/or augment the functionality of debs. We use it with the Slurm snap and the `slurm_ops` charm library to make it really easy to exchange keys between machines without having to get clever with the Slurm protocol.

Given that it can be used outside of the Slurm snap to support general `munge` operations (and makes it really easy for us to set up keys), it should be put into its own repository. Also, by spinning `mungectl` out into it's own repository, it enables the Slurm snap repository to be more focused.

## Other motivations

* Gain practice and experience with creating a Debian package from scratch for a Golang project.

**Comment by jedel1043** [comment]
I think this sounds like a great idea! It should make it easier to maintain in the long run, and could be useful for some other applications that use Munge without using Slurm.

**Reply by NucciTheBoss** [reply]
Okay, great! I'll go ahead a spin up a repo then. After that I'll fork it so I can open a PR for the work I'm doing to make a Debian package for `mungectl`

**Reply by NucciTheBoss** [reply]
Spun out `mungectl` into its own repo and have a proposed Debian package recipe for it. Once we polish it a bit further - like add a README and CONTRIBUTING guidelines - we can do the work to drop the embedded version from the Slurm snap and replace it with the new repo version.

**Reply by jamesbeedy** [reply]
sweet!

**Comment by NucciTheBoss** [comment]
Merged PR to add README and contributing guidelines to `mungectl`. As such, I consider this discussion resolved!

https://github.com/charmed-hpc/mungectl/pull/4
